server/consumer.go

```diff
@@ -2587,6 +2580,21 @@ func (o *consumer) processNextMsgRequest(reply string, msg []byte) {
 	}
 }
 
+func trackDownAccountAndInterest(acc *Account, interest string) (*Account, string) {
+	for done := false; !done && strings.HasPrefix(interest, replyPrefix); {
```

Comment:
Reads funny..

Reply:
I can remove the done from the for-loop initialization and define it inside the loop and break if true. Let me see if it is better..

Reply:
Since it is just a matter of taste, please select version 1, 2 or 3, or propose a 4 :-)
1:
```
	for strings.HasPrefix(interest, replyPrefix) {
		done := true // will be set to false if we need to continue the loop
		oa := acc
		oa.mu.RLock()
		if oa.exports.responses != nil {
			if si := oa.exports.responses[interest]; si != nil {
				acc, interest, done = si.acc, si.to, false
			}
		}
		oa.mu.RUnlock()
		if done {
			break
		}
	}
	return acc, interest
```
2:
```
         for done := false; !done; {
		if !strings.HasPrefix(interest, replyPrefix) {
			break
		}
		oa := acc
		oa.mu.RLock()
		if oa.exports.responses == nil {
			done = true
		} else if si := oa.exports.responses[interest]; si != nil {
			acc, interest = si.acc, si.to
		} else {
			done = true
		}
		oa.mu.RUnlock()
	}
	return acc, interest
```
3:
```
	for strings.HasPrefix(interest, replyPrefix) {
		oa := acc
		oa.mu.RLock()
		if oa.exports.responses == nil {
			oa.mu.RUnlock()
			break
		}
		si := oa.exports.responses[interest]
		if si == nil {
			oa.mu.RUnlock()
			break
		}
		acc, interest = si.acc, si.to
		oa.mu.RUnlock()
	}
	return acc, interest
```

Reply:
Went with approach 3.